### ISSUE TITLE
Unable to verify server certificate

### ISSUE THREAD

**Issue by aleksandr-kovytin-incountry** [wazuh/wazuh#13772]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.3.2-1 | wazuh-agent | Agent | Packages | Ubuntu 18.04.6 LTS |

I have a problem with manager identity verification during auto-enrollment of an agent.

Here's the debug log on the agent:
```
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":477,"routine":"w_enrollment_verify_ca_certificate","level":"info","description":"Verifying manager's cer
tificate"}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"check_cert.c","line":52,"routine":"check_x509_cert","level":"debug","description":"Checking certificate's subject alternative na
mes."}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"check_cert.c","line":58,"routine":"check_x509_cert","level":"debug","description":"No matching subject alternative names found. 
Checking common name."}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":479,"routine":"w_enrollment_verify_ca_certificate","level":"error","description":"Unable to verify server certificate"}
```
The last row shows that the agent couldn't verify the manager identity. However, the next rows show that the agent is enrolled despite of that:
```
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":589,"routine":"w_enrollment_load_pass","level":"info","description":"Using password specified on file: /var/ossec/etc/authd.pass"}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":267,"routine":"w_enrollment_send_message","level":"info","description":"Using agent name as: nonprod-tools-aws-se-clamav-mirror-10-9-12-191"}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":307,"routine":"w_enrollment_send_message","level":"debug","description":"Request sent to manager"}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":332,"routine":"w_enrollment_process_response","level":"info","description":"Waiting for server reply"}
{"timestamp":"2022/06/07 13:48:39","tag":"wazuh-agentd","pid":24853,"file":"enrollment_op.c","line":453,"routine":"w_enrollment_process_agent_key","level":"info","description":"Valid key received"}
```
 
I tried to enroll agent using agent-auth, and it went smoothly
```
# /var/ossec/bin/agent-auth -m wazuh.tools.example.com -v /var/ossec/etc/manager_rootCA.pem
2022/06/07 10:24:11 agent-auth: INFO: Started (pid: 15874).
2022/06/07 10:24:11 agent-auth: INFO: Requesting a key from server: wazuh.tools.example.com
2022/06/07 10:24:11 agent-auth: INFO: Verifying manager's certificate
2022/06/07 10:24:11 agent-auth: INFO: Manager has been verified successfully
2022/06/07 10:24:11 agent-auth: INFO: Using password specified on file: etc/authd.pass
2022/06/07 10:24:11 agent-auth: INFO: Using agent name as: nonprod-tools-aws-se-clamav-mirror-10-9-12-191
2022/06/07 10:24:11 agent-auth: INFO: Waiting for server reply
2022/06/07 10:24:11 agent-auth: INFO: Valid key received
```

I also tried to debug with
```
openssl s_client -connect wazuh.tools.example.com:1515 -CAfile /var/ossec/etc/manager_rootCA.pem
```
but there were no errors neither.

**So there are actually two issues:
1. Inability of the agent to verify the manager during auto-enrollment
2. The agent is enrolled despite of manager’s verification fail**

Enrollment config on the agent is
```
  <enrollment>
      <enabled>yes</enabled>
      <manager_address>wazuh.tools.example.com</manager_address>
      <port>1515</port>
      <server_ca_path>/var/ossec/etc/manager_rootCA.pem</server_ca_path>
      <authorization_pass_path>/var/ossec/etc/authd.pass</authorization_pass_path>
      <auto_method>no</auto_method>
      <delay_after_enrollment>20</delay_after_enrollment>
      <use_source_ip>no</use_source_ip>
    </enrollment>
```

The config on the manager's side is 
```
  <auth>
    <disabled>no</disabled>
    <port>1515</port>
    <use_source_ip>no</use_source_ip>
    <force>
      <enabled>yes</enabled>
      <key_mismatch>yes</key_mismatch>
      <disconnected_time enabled="yes">1h</disconnected_time>
      <after_registration_time>1h</after_registration_time>
    </force>
    <purge>yes</purge>
    <use_password>yes</use_password>
    <ciphers>HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH</ciphers>
    <ssl_verify_host>no</ssl_verify_host>
    <ssl_manager_cert>/var/ossec/etc/sslmanager.cert</ssl_manager_cert>
    <ssl_manager_key>/var/ossec/etc/sslmanager.key</ssl_manager_key>
    <ssl_auto_negotiate>no</ssl_auto_negotiate>
  </auth>
```

**Comment by vikman90** [comment]
@aleksandr-kovytin-incountry Thank you very much for reporting this problem.

I've reproduced this case and I confirm that, as of version 3.13, the agent prints an error when it fails to verify the manager, but it continues to enroll.

We will address this issue ASAP and will keep you posted.

Thank you again.

**Comment by vikman90** [comment]
@aleksandr-kovytin-incountry this issue has been fixed by #13907 and will apply in the next patch version (4.3.5).

Thank you again for reporting this problem!

Best.
